docs/contributing/versioning.md

```diff
@@ -0,0 +1,121 @@
+
+- you think you can sneak a change in with other breaking changes - breaking changes need to be migrated from, which is effort for our users we should try to avoid
+- changing the name of an API, based on a hunch
+- adopting a technology based on popularity and not because of the problems they solve
+- changing a components interface without a good reason to do so
+
+This is similar to how we try to tackle most problems, by focusing on user needs first.
+
+Ember.JS is a great example of prioritising stability see [Stability without Stagnation: Lessons Learned from Shipping Ember.js
+](https://speakerdeck.com/wycats/stability-without-stagnation-lessons-learned-from-shipping-ember-dot-js)
+
+We do not want our users to be reliant on certain versions of our library with no way forward.
+
+Whenever we decide to make a breaking change we must ensure we do our best to try to first deprecate the feature and allow a migration path.
+
+## Proposal
+
+Some breaking changes that need discussion may first be proposed in the [GOV.UK Design System architecture repository](https://github.com/alphagov/govuk-design-system-architecture/blob/master/proposals/README.md).
+
+This is to ensure the community can get involved with the decision.
+
+You should make sure to take steps to actively involved the community and not wait for feedback, this might be in the form of presentations or meetings. Since these issues may be difficult to understood without a conversation.
+
+## Deprecation
+
+Deprecation is the practice of making breaking changes gradually, and is part of communicating upcoming changes
+that would require a user to update their application.
+
+Deprecating parts of the library help users migrate to the new way while still being able to run the old code.
+
+Note: Our users will may not know what 'deprecation' means, so it's important to also clarify in full what this means.
+
+Example 1: Fixing a typo in a CSS class name.
+
+1. We discover the class name `.govuk-visually-hidden-focussable` includes the typo 'focussable'
+2. We raise a pull request that renames the class to `.govuk-visually-hidden-focusable` while keeping
+the previous class available.
+3. We add a comment to the source code that indicates this is deprecated, and raise an issue to remove it in a future breaking release.
+4. When releasing the change we include a clear summary that indications what was the problem, what we've changed and how a user can migrate before the future breaking release.
+
+Sometimes it is not possible to deprecate code, this is OK but try to make this a last resort.
+
+Migrating takes a lot of effort for our users, and if done badly can result in a loss of trust.
+
+Do not use a breaking release of deprecated features as an opportunity to make breaking changes based on the ‘bad reasons to make breaking changes’ in the ‘Stability’ section.
+
+## Migration
+
+Migration is the practice of a user moving from one approach to an equivalent approach.
```

Comment:
I would be specific that this is how you're defining it here. Some definitions would be more specific than simply "approach" e.g. "OS migration".